### ISSUE TITLE
improve speed of "parallel by file" by somehow pruning instances early

### ISSUE THREAD

**Issue by ethangodt** [grawk/nemo-runner#17]
I think because "parallel by file" spawns a new process for every file (even before we know if those files need to be tested) it's often very slow. In our use case we'll want to run 6 different profiles against some tagged section of tests. We currently have 61 files of tests (one test per file). Even if we want to run just one file, nemo-runner would spawn 366 child processes to begin checking which single test should be run. This adds several minutes before things get going.

Possible solutions:
- Find some way to prune instances before child processes are created, so only instances with tests to run get parallel child processes. My intuition is that looping through all the instances and [manually checking whether mocha would even run ](https://github.com/paypal/nemo-runner/pull/13/files#diff-a5b70841c2143e5d2e0fd98dd3b4cefeR67) would speed things up considerably. (Can we pass our mocha instance to the child process? I don't have experience with this.)

**Comment by grawk** [comment]
Good catch. Thanks for trying to wring out performance improvements. I agree it's problematic to spawn child processes that may not actually run.
